static/downloadables/parse_associations.py

```diff
@@ -74,20 +75,81 @@ def getDownloadedFiles(fileHash, requiredParamsHash, superPop, mafCohort, refGen
         # open the files that were previously created
         with open(associationsPath, 'r') as tableObjFile:
             tableObjDict = json.load(tableObjFile)
-        with open(clumpsPath, 'r') as clumpsObjFile:
-            clumpsObjDict = json.load(clumpsObjFile)
         with open(clumpNumPath, 'r') as clumpNumFile:
             clumpNumDict = json.load(clumpNumFile)
         with open(studySnpsPath, 'r') as studySnpsFile:
             studySnpsDict = json.load(studySnpsFile)
         with open(mafCohortPath, 'r') as mafFile:
             mafDict = json.load(mafFile)
+
+	# Get super populations from studyIDMetaData
+        allSuperPops = set()
+        for study in tableObjDict['studyIDsToMetaData']:
+            for trait in tableObjDict['studyIDsToMetaData'][study]['traits']:
+                superPopList = tableObjDict['studyIDsToMetaData'][study]['traits'][trait]['superPopulations']
+                preferredPop = getPreferredPop(superPopList, superPop)
+                allSuperPops.add(preferredPop)
+	
+	# loop through each population and download the corresponding clumps file. Add that file to a dictionary
+	# where {pop:clumps object, pop2: clumps object, etc.}
+        print(str(allSuperPops))
+        allClumps = {}
+        for pop in allSuperPops:
```

Comment:
My original thought with having them be saved by super population into separate files was to load them separately. If we are going to load them into one object, then maybe we should just save it in one file. If it is filtered, we can just call it clumps_{r}\_{ahash}.txt, for all the clumps, allClumps\_{r}.txt

Reply:
That was my original thought, but I think that that could potentially cause some issues with the getAllClumps function (and corresponding js functions) in connect_to_server.py. If I understand correctly, we would have to have a file on the server that includes all populations--which would be a really large (potentially problematic) file.
I don't understand the server side of things as much, but it seems like if we were to create a combined file, it would require re-doing a bunch of the js scripts, right?

Reply:
That is very true. Lets just keep it like this then